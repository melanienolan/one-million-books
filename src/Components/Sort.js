import React from 'react';
import PropTypes from 'prop-types';
import ArrowUp from 'react-icons/lib/fa/arrow-up';
import ArrowDown from 'react-icons/lib/fa/arrow-down';

const Sort = ({ type, sortBooks }) => {
  const capitalizeFirstLetter = str => {
    return str.charAt(0).toUpperCase() + str.slice(1);
  };
  return (
    <div className="sort--holder">
      <span className="sort--title">
        {capitalizeFirstLetter(type)}
      </span>
      <button className="button--sort" onClick={() => sortBooks(type, 1)}>
        <ArrowDown />
      </button>
      <button className="button--sort" onClick={() => sortBooks(type, 0)}>
        <ArrowUp />
      </button>
    </div>
  );
};

export default Sort;

Sort.propTypes = {
  type: PropTypes.string,
  sortBooks: PropTypes.func
};
