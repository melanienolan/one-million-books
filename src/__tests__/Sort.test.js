import React from 'react';
import Sort from '../Components/Sort';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

it('renders sort component without crashing', () => {
  Enzyme.shallow(<Sort type="title" />);
});
it('renders sort component with title', () => {
  const sortComponent = Enzyme.shallow(<Sort type="title" />);
  expect(toJson(sortComponent)).toMatchSnapshot();
});
it('renders sort component with id', () => {
  const sortComponent = Enzyme.shallow(<Sort type="id" />);
  expect(toJson(sortComponent)).toMatchSnapshot();
});
