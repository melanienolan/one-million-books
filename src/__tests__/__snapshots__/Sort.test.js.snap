// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sort component with id 1`] = `
<div
  className="sort--holder"
>
  <span
    className="sort--title"
  >
    Id
  </span>
  <button
    className="button--sort"
    onClick={[Function]}
  >
    <FaArrowDown />
  </button>
  <button
    className="button--sort"
    onClick={[Function]}
  >
    <FaArrowUp />
  </button>
</div>
`;

exports[`renders sort component with title 1`] = `
<div
  className="sort--holder"
>
  <span
    className="sort--title"
  >
    Title
  </span>
  <button
    className="button--sort"
    onClick={[Function]}
  >
    <FaArrowDown />
  </button>
  <button
    className="button--sort"
    onClick={[Function]}
  >
    <FaArrowUp />
  </button>
</div>
`;
